projects/s-rxjs-utils/src/lib/map-with-caching.ts

```diff
@@ -0,0 +1,46 @@
+import { ObjectWith } from "micro-dash";
+import { map } from "rxjs/operators";
+
+/**
+ * Applies a given function to each value in the array emitted by the source Observable, and emits the resulting array as an Observable. Each item in the array is cached using key generated by the given function so that the resulting arrays contain references to the exact same objects included in the previous emission, when possible.
+ *
+ * @param buildCacheKey A function that converts an upstream object into a string to use as the cache key. Needs to return a unique key for each item in the source array.
+ * @param buildDownstreamItem A function that converts an upstream object into a downstream object
```

Comment:
`[green]`
I don't think this is very useful. I would leave documentation for this param out since it basically just repeats the name and type info.